ReactAndroid/src/main/java/com/facebook/react/devsupport/MultipartStreamReader.java

```diff
@@ -113,7 +148,10 @@ public boolean readAllParts(ChunkCallback callback) throws IOException {
         Buffer chunk = new Buffer();
         content.skip(chunkStart);
         content.read(chunk, length);
+        emitProgress(currentHeaders, chunk.size() - currentHeadersLength, true, progressCallback);
```

Comment:
true -> isCloseDelimiter?

Reply:
We do emit progress for each chunk so even if it is not the last chunk (what isCloseDelimiter means) we still want to pass isFinal to true. It's only use is to make sure the last event isn't throttled so we see 100%.

Reply:
If I understand this correctly, we'd reach 100% only on the last chunk -> don't we want to force the progress event only in that scenario then? Otherwise I'd say `isFinal` should be named something else if it gets set to true many times and not just for the final progress event.

Reply:
No it is not the case, progress is reset for each chunk so let's say we have three chunks progress would look like: 0, 1, 0, 1, 0, 1. In the case of RN packager we only care about the progress of the last chunk (the JS bundle so we ignore progress of the other ones (https://github.com/facebook/react-native/pull/17809/files/bbe67864421b1de228d7af3ddd7327ed8bca060c#diff-948206404ada96979d687d628e935ee1R204).

Reply:
Just to clarify a chunk is a "part" from the multipart response. You can pretty much see it as multiple http responses nested inside the normal http response. It has it's own headers and we track progress of downloading each chunk separately.

Reply:
I think I get it. I thought that the progress was for the entire multipart response (e.g. like a multipart upload) rather than for just the chunk.

What do you think about refactoring the ChunkCallback interface so that it's clear the progress events are just for that one chunk? (Breaking change to the API)

```java
  public interface ChunkListener {
    /**
     * Invoked when a chunk of a multipart response is fully downloaded.
     */
    void onChunkComplete(Map<String, String> headers,
                         Buffer body,
                         boolean isLastChunk) throws IOException;

    /**
     * Invoked as bytes of the current chunk are read and may be invoked multiple times per chunk.
     * This method is guaranteed to be invoked one last time when the chunk is fully downloaded,
     * with `isChunkComplete` set.
     */
     void onChunkProgress(Map<String, String> headers,
                          long chunkBytesRead,
                          long chunkTotalBytes,
                          boolean isChunkComplete) throws IOException;
  }
```

Reply:
Makes sense, I agree it will be clearer like that, not sure if we have to mark this as breaking since I don't think it is meant to be public api.